### ISSUE TITLE
Confused about v-prediction SDS loss

### ISSUE THREAD

**Issue by HeliosZhao** [threestudio-project/threestudio#325]
Hi, thanks a lot for the brilliant work.

In the implementation of SDS loss with v-prediction ([here](https://github.com/threestudio-project/threestudio/blob/607a4be985be817330b8ef79f817da6f255c14e5/threestudio/models/guidance/stable_diffusion_vsd_guidance.py#L515)). The final grad is 
$|\epsilon-\sqrt{1-\alpha_t}z_0-\sqrt{\alpha_t}\epsilon_\theta|$, where $\epsilon$ and $\epsilon_\theta$ denotes the random noise and the model predicted noise. 

However, in the training loss of v-prediction diffusion model. For example, [in train_lora method](https://github.com/threestudio-project/threestudio/blob/607a4be985be817330b8ef79f817da6f255c14e5/threestudio/models/guidance/stable_diffusion_vsd_guidance.py#L563). The loss is formulated as the below equation as [here](https://github.com/threestudio-project/threestudio/blob/607a4be985be817330b8ef79f817da6f255c14e5/threestudio/models/guidance/stable_diffusion_vsd_guidance.py#L584) and [here](https://github.com/huggingface/diffusers/blob/e5168588864d72a4dca37e90318c6b11da0eaaf1/src/diffusers/schedulers/scheduling_ddpm.py#L478). The loss is:
$|\sqrt{\alpha_t}\epsilon-\sqrt{1-\alpha_t}z_0-\epsilon_\theta|$, where $\epsilon$ and $\epsilon_\theta$ denotes the random noise and the model predicted noise. 

As in DreamFusion, we should use the diffusion loss as the grad. Therefore, I am not sure the v-prediction SDS loss is correct or not.

**Comment by DSaurus** [comment]
Hi @HeliosZhao ,

Actually, the correct expression for the final gradient is not $|\epsilon-\sqrt{1-\alpha_t}z_0-\sqrt{\alpha_t}\epsilon_\theta|$, but rather $|\epsilon-\sqrt{1-\alpha_t}z_t-\sqrt{\alpha_t}\epsilon_\theta|$. Note that $z_t$ is not equal to $z_0$. So the original implementation is correct.

**Comment by HeliosZhao** [comment]
Hi @DSaurus ,
Thanks for the reply. Please note that my question is not about $z_t$ and $z_0$. 

My question is, the expression for v-prediction training in diffusion is different from the SDS loss in your implementation.
The v-prediction in diffusion training is:
$|\sqrt{\alpha_t}\epsilon-\sqrt{1-\alpha_t}z_t-\epsilon_\theta|$

In your SDS implementation, the v-prediction SDS loss is:
$|\epsilon-\sqrt{1-\alpha_t}z_t-\sqrt{\alpha_t}\epsilon_\theta|$

The coefficients for $\epsilon$ and $\epsilon_\theta$ are different.

**Comment by DSaurus** [comment]
Actually the v-prediction in diffusion training is $|\sqrt{\alpha_t}\epsilon-\sqrt{1-\alpha_t}z_0-\epsilon_\theta|$.  Note that $z_0$ is not equal to $z_t$. You can use it to derive the SDS loss $|\epsilon-\sqrt{1-\alpha_t}z_t-\sqrt{\alpha_t}\epsilon_\theta|$.

**Comment by HeliosZhao** [comment]
Oh. I got it. Thanks!
